pkg/reconciler/v1alpha1/taskrun/entrypoint/entrypoint.go

```diff
@@ -76,116 +83,151 @@ func (c *Cache) set(sha string, ep []string) {
 	c.lru.Add(sha, ep)
 }
 
+// AddToEntrypointCache adds an image digest and its entrypoint
+// to the cache
+func AddToEntrypointCache(c *Cache, sha string, ep []string) {
+	c.set(sha, ep)
+}
+
 // AddCopyStep will prepend a BuildStep (Container) that will
 // copy the entrypoint binary from the entrypoint image into the
 // volume mounted at MountPoint, so that it can be mounted by
 // subsequent steps and used to capture logs.
-func AddCopyStep(ctx context.Context, b *v1alpha1.BuildSpec) {
-	cfg := config.FromContext(ctx).Entrypoint
+func AddCopyStep(b *v1alpha1.BuildSpec) {
 
 	cp := corev1.Container{
-		Name:         InitContainerName,
-		Image:        cfg.Image,
-		Command:      []string{"/bin/cp"},
-		Args:         []string{"/entrypoint", BinaryLocation},
+		Name:    InitContainerName,
+		Image:   *entrypointImage,
+		Command: []string{"/bin/sh"},
+		// based on the ko version, the binary could be in one of two different locations
+		Args:         []string{"-c", fmt.Sprintf("if [[ -d /ko-app ]]; then cp /ko-app/entrypoint %s; else cp /ko-app %s;  fi;", BinaryLocation, BinaryLocation)},
 		VolumeMounts: []corev1.VolumeMount{toolsMount},
 	}
 	b.Steps = append([]corev1.Container{cp}, b.Steps...)
 
 }
 
-type entrypointArgs struct {
-	Args       []string `json:"args"`
-	ProcessLog string   `json:"process_log"`
-	MarkerFile string   `json:"marker_file"`
+// RedirectSteps will modify each of the steps/containers such that
+// the binary being run is no longer the one specified by the Command
+// and the Args, but is instead the entrypoint binary, which will
+// itself invoke the Command and Args, but also capture logs.
+func RedirectSteps(cache *Cache, steps []corev1.Container, kubeclient kubernetes.Interface, build *buildv1alpha1.Build, logger *zap.SugaredLogger) error {
+	for i := range steps {
+		step := &steps[i]
+		if len(step.Command) == 0 {
+			logger.Infof("Getting Cmd from remote entrypoint for step: %s", step.Name)
+			var err error
+			step.Command, err = GetRemoteEntrypoint(cache, step.Image, kubeclient, build)
+			if err != nil {
+				logger.Errorf("**ALERT**: Error getting entry point image", err.Error())
```

Comment:
does this error mean:
- the controller has a problem <-- this is what the message currently implies
- the user may have made an error (problem with image, problem with credentials)

if it's more likely the second, i dont think we should log something at all - if we do, i think it should be a bit less urgent looking

Reply:
I'll remove the alert part of the message, but I think its useful to log something when we couldn't get the remote image, wdyt?